### ISSUE TITLE
Wrong turn in a junction

### ISSUE THREAD

**Issue by yotamN** [osmandapp/Osmand#7298]
On version 3.4.4 of OSMAnd+ I tried to route through [Kiryat Shmona][0] and I got a weird turn. If you will try to drive for example from [Metula][1] to [Rosh Pina][2] you will see that at [Barzilai street][3] you are told to turn right and then immediately left again, like so:
![Screenshot_20190720-212022_OsmAnd~](https://user-images.githubusercontent.com/5249239/61582536-c4ce1500-ab34-11e9-91f2-c61844e43a12.jpg)

Of course the expected behaviour would be to continue straight and that's what GraphHopper and OSRM are showing on OpenStreetMap.

[0]: https://www.openstreetmap.org/way/578640475
[1]: https://www.openstreetmap.org/way/69671571
[2]: https://www.openstreetmap.org/way/696713262
[3]: https://www.openstreetmap.org/way/352638071

**Comment by naoliv** [comment]
It seems that the `trunk_link` ways (like https://www.openstreetmap.org/way/333770262) are missing `maxspeed`, like this way https://www.openstreetmap.org/way/333770262

**Comment by scaidermern** [comment]
And using the link road avoids the traffic lights. Although it requires crossing a junction, maybe this penalty is too low or the traffic signal penalty too high?

**Comment by vshcherb** [comment]
I think we are not going to fix it unless we clarify that mapping is correct:
1. No missing max_speed on links
2. Either missing traffic_lights or give_away sign to come back to motorway. It looks really strange with traffic signals in between so from a routing engine it could make sense to have such deviation  but I'm pretty sure it is not allowed even if you want to do it.

**Comment by yotamN** [comment]
> 1. No missing max_speed on links

The default speed limit in a city is 50, no?  
But there are a lot of roads with missing speed limit, wouldn't it be better to try to handle it correct too?

> 2. Either missing traffic_lights or give_away sign to come back to motorway. It looks really strange with traffic signals in between so from a routing engine it could make sense to have such deviation but I'm pretty sure it is not allowed even if you want to do it.

Again, some signs are missing, I will check next time I'm there if the is a sign but I reported it because it's not the only place that will miss a sign. Other routers that I tried didn't made me do the turn so I hope it can be fixed.  
Maybe it doesn't take into account the hard turn and coming back to the road.
